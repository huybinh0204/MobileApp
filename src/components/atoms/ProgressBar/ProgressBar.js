import PropTypes from 'prop-types';
import React, { useEffect, useRef } from 'react';
import { Animated, Easing } from 'react-native';
import { Bar, Progress } from './ProgressBar.styles';

const ProgressBar = ({ step, numberOfSteps }) => {
  const prevStep = Math.min(0, step - 1);
  const { current: progress } = useRef(new Animated.Value(prevStep));

  const width = progress.interpolate({
    inputRange: [0, numberOfSteps],
    outputRange: ['0%', '100%'],
    extrapolate: 'clamp',
  });

  useEffect(() => {
    Animated.timing(progress, {
      easing: Easing.ease,
      duration: 600,
      toValue: step,
      useNativeDriver: false,
    }).start();
  }, [progress, step]);

  return (
    <Bar>
      <Progress style={{ width }} />
    </Bar>
  );
};

ProgressBar.propTypes = {
  step: PropTypes.number.isRequired,
  numberOfSteps: PropTypes.number.isRequired,
};

export default ProgressBar;
