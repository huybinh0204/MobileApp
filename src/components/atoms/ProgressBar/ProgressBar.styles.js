import styled from '@emotion/native';
import { Animated, StyleSheet } from 'react-native';
import { normalize } from '_utilities/screen';

export const Bar = styled.View(({ theme }) => ({
  alignSelf: 'center',
  backgroundColor: theme.colors.beta50,
  borderColor: theme.colors.black,
  borderRadius: 10,
  height: normalize(5),
  width: '100%',
}));

export const Progress = styled(Animated.View)(({ theme }) => ({
  ...StyleSheet.absoluteFillObject,
  backgroundColor: theme.colors.alpha500,
  borderRadius: 10,
}));
