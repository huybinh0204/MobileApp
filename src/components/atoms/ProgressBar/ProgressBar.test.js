import { render } from '@testing-library/react-native';
import React from 'react';
import Theme from '_utilities/Theme';
import ProgressBar from './ProgressBar';

describe('ProgressBar', () => {
  it('should match the snapshot', () => {
    const { toJSON } = render(
      <Theme>
        <ProgressBar step={2} numberOfSteps={10} />
      </Theme>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
