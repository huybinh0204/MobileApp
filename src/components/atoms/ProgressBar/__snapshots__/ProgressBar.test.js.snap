// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "backgroundColor": "#F2F3F4",
        "borderColor": "#000000",
        "borderRadius": 10,
        "height": 9,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "#3D3397",
        "borderRadius": 10,
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": "0%",
      }
    }
  />
</View>
`;
